cognite/powerops/resync/v1/data_classes.py

```diff
@@ -0,0 +1,449 @@
+        self.name = name
+        self.type_ = type_
+        self.properties = properties
+        self.property_configurations = property_configurations
+
+    @classmethod
+    def from_yaml(cls, configuration_path: Path) -> dict[str, DataModelConfiguration]:
+        """Generate a dictionary of DataModelConfiguration objects from a YAML file.
+
+        Given a path to a YAML file containing data model configurations, this function generates a dictionary of
+        DataModelConfiguration objects.
+
+        Args:
+            configuration_path: The path to the YAML file containing data model configurations.
+
+        Returns:
+            A dictionary of DataModelConfiguration objects.
+
+        Raises:
+            ValueError: If a property configuration is not a valid property for the domain model type.
+        """
+        raw_data_model_configuration = load_yaml(configuration_path, expected_return_type="dict")
+
+        all_write_classes = get_data_model_write_classes()
+
+        all_data_model_configurations = {}
+        for name, type_ in all_write_classes.items():
+            properties = get_properties_for_type(type_)
+            property_configurations = PropertyConfiguration.from_dictionary(raw_data_model_configuration.get(name, {}))
+
+            # check if all property configurations are valid properties
+            for property_configuration in property_configurations:
+                if property_configuration.property not in properties:
+                    # TODO: make logging??
+                    raise ValueError(
+                        f"Property {property_configuration.property} not in domain model properties for {name}"
+                    )
+
+            all_data_model_configurations[name] = cls(
+                type_=type_, name=name, properties=properties, property_configurations=property_configurations
+            )
+
+        return all_data_model_configurations
+
+
+@dataclass
+class ResyncConfiguration:
+    """Data class for resync configuration.
+
+    Attributes:
+        data_model_configuration_file: The path to the data model configuration file.
+        working_directory: The working directory for the resync importer.
+        logging_level: The logging level for the resync importer.
+        overwrite_data: Whether to overwrite existing data in CDF.
+        data_set_id: The data set id to use for the resync importer.
+        all_write_classes: A dictionary of all domain model write classes.
+        file_configuration: A FileUploadConfiguration object detailing the file upload configuration.
+    """
+
+    data_model_configuration_file: Path
+    working_directory: Path
+    overwrite_data: bool
+    data_set_id: int
+    all_write_classes: dict[str, type]
+    file_configuration: Optional[FileUploadConfiguration]
+
+    def __init__(
+        self,
+        data_model_configuration_file: str,
+        working_directory: str,
+        data_set_id: int,
```

Comment:
Can we use `data_set_external_id: str` instead? (And find out where it makes sense to do `data_set_id = client.data_set.retrieve(external_id=resync_config.data_set_external_id).id`...)

I think this will make it quite a bit easier to specify, in particular if we want to use the same config for deploying to different environments.